src/models/decomposition_models.jl

```diff
@@ -168,14 +168,19 @@ $ICA_DESCR
 end
 
 function MMI.fit(model::ICA, verbosity::Int, X)
+    icagfun(fname::Symbol, ::Type{T} = Float64) where T<:Real=
+    fname == :tanh ? MS.Tanh{T}(1.0) :
+    fname == :gaus ? MS.Gaus{T}() :
+    error("Unknown gfun $(fname)")
+
```

Comment:
@testercwt I've just had a belated realisation that this should be handled differently. Ordinarily, checks for the validity of hyper-parameters (fields of the model struct) are not put in `fit!`. Rather they live in a [`clean!` method](https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use/#New-model-type-declarations-and-optional-clean!-method) which, in this case, has been generated automatically by the `@mlj_model` macro, and line 162. So you remove this check. 

MLJ users interact via the machine interface, which ensures `clean!` is called before `fit` or `update` every time.

Reply:
There's always a better way. But I have no idea which is the better one. 
Since icagfun,:tanh,:gaus are no longer exist in MS 0.9.1 , the straight forward thoughts is to abandon  icagfun,:tanh,:gaus as well. However, @mlj_model cannot handle parametric type and I think change the struct ICA definition is not a good idea  at present.

Reply:
new PR preferred for backwards compatability , IMO.